lib/FundDistribution/FundDistributionView/FundDistributionViewContainer.js

```diff
@@ -1,30 +1,64 @@
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+} from 'react';
 import PropTypes from 'prop-types';
-import { get } from 'lodash';
 
 import {
   stripesConnect,
 } from '@folio/stripes/core';
 
-import { fundsManifest } from '../../manifests';
+import { fundsManifest, transactionsManifest } from '../../manifests';
 import FundDistributionView from './FundDistributionView';
 import { DICT_FUNDS } from '../../constants';
 import { fundDistributionShape } from '../../shapes';
+import { batchFetch } from '../../utils';
 
 const FundDistributionViewContainer = ({
-  resources, totalAmount, currency, fundDistributions, stripes,
+  mutator, totalAmount, currency, fundDistributions, stripes,
 }) => {
-  if (!get(resources, [DICT_FUNDS, 'hasLoaded'])) return null;
-  const funds = get(resources, [DICT_FUNDS, 'records'], []);
-  const fundsToDisplay = fundDistributions.map(distr => {
-    const { name = '', code = '' } = funds.find(({ id }) => id === distr.fundId) || {};
-    const fundName = `${name}${code ? `(${code})` : ''}`;
-
-    return {
-      ...distr,
-      fundName,
-    };
-  });
+  const [isLoading, setIsLoading] = useState(true);
+  const [funds, setFunds] = useState();
+  const [fundsToDisplay, setFundsToDisplay] = useState([]);
+
+  useEffect(
+    () => {
+      setIsLoading(true);
+      setFundsToDisplay([]);
+
+      const fundsPromise = funds ? Promise.resolve(funds) : mutator[DICT_FUNDS].GET();
+      const encumbrancesPromise = batchFetch(
+        mutator.fundDistributionEncumbrances,
+        fundDistributions
+          .filter(({ encumbrance }) => encumbrance)
+          .map(({ encumbrance }) => encumbrance),
+      );
+
+      Promise.all([fundsPromise, encumbrancesPromise])
+        .then(([fundsResponse, encumbrancesResponse]) => {
+          setFunds(fundsResponse);
+
+          setFundsToDisplay(fundDistributions.map(distr => {
+            const { name = '', code = '' } = fundsResponse.find(({ id }) => id === distr.fundId) || {};
+            const fundName = `${name}${code ? `(${code})` : ''}`;
+            const fundEncumbrance = encumbrancesResponse.find(({ id }) => id === distr.encumbrance);
+
+            return {
+              ...distr,
+              fundName,
+              fundEncumbrance,
+            };
+          }));
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [fundDistributions],
```

Comment:
this is a result of `adjustments.reduce()`, so basically it makes call to the api every render